### PR TITLE
HUM-904 Add put/get/del check for hummingbird

### PR DESCRIPTION
**Pull request by corystone**

### CODE REVIEW

File: playbooks/files/rax-maas/plugins/hummingbird_monitoring.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+
```
Comment (tonytan4ever) [inline playbooks/files/rax-maas/plugins/hummingbird_monitoring.py]: These are PEP8 failures:
```
checking /var/lib/jenkins/workspace/PR_rpc-maas-master-container-pep8-tox-test/rpc-maas/playbooks/files/rax-maas/plugins/hummingbird_monitoring.py
/var/lib/jenkins/workspace/PR_rpc-maas-master-container-pep8-tox-test/rpc-maas/playbooks/files/rax-maas/plugins/hummingbird_monitoring.py:17:1: F401 'argparse' imported but unused
/var/lib/jenkins/workspace/PR_rpc-maas-master-container-pep8-tox-test/rpc-maas/playbooks/files/rax-maas/plugins/hummingbird_monitoring.py:19:1: F401 'pprint' imported but unused
/var/lib/jenkins/workspace/PR_rpc-maas-master-container-pep8-tox-test/rpc-maas/playbooks/files/rax-maas/plugins/hummingbird_monitoring.py:22:1: F401 'get_auth_details' imported but unused
/var/lib/jenkins/workspace/PR_rpc-maas-master-container-pep8-tox-test/rpc-maas/playbooks/files/rax-maas/plugins/hummingbird_monitoring.py:24:1: F811 redefinition of unused 'get_keystone_client' from line 23
/var/lib/jenkins/workspace/PR_rpc-maas-master-container-pep8-tox-test/rpc-maas/playbooks/files/rax-maas/plugins/hummingbird_monitoring.py:53:9: F841 local variable 'etag' is assigned to but never used
/var/lib/jenkins/workspace/PR_rpc-maas-master-container-pep8-tox-test/rpc-maas/playbooks/files/rax-maas/plugins/hummingbird_monitoring.py:77:1: W391 blank line at end of file
```

Comment (corystone) [reply playbooks/files/rax-maas/plugins/hummingbird_monitoring.py]: Thanks, figured out tox lint!

File: playbooks/templates/rax-maas/hummingbird_transaction.yaml.j2 [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+disabled    : "false"
+details     :
+    file    : run_plugin_in_venv.sh
+    args    : ["/usr/lib/rackspace-monitoring-agent/plugins/hummingbird_monitoring.py"]
+    timeout : 59000
+{{ get_metadata(label).strip() }}
+alarms      :
+    hummingbird_transaction_alarm:
+        label                   : hummingbird_transaction_alarm
+        notification_plan_id    : "npTechnicalContactsEmail"
```
Comment (npawelek) [inline playbooks/templates/rax-maas/hummingbird_transaction.yaml.j2]: This will need to defined as `maas_notification_plan`

**Comment by npawelek** [comment]
@creiht can you confirm with the `Confd` lines of the agent log (`/var/log/rackspace-monitoring-agent.log`) after restarting the agent? I've noticed some issues recently where these lines will be non-existent and the agent doesn't necessarily complain.

**Comment by creiht** [comment]
@npawelek I'm not 100% sure what you are wanting to see but I did see this: 

`Wed Jul 18 08:05:12 PM 2018 INF: Confd -> config_file post operation result: success for alarm at unchanged, handle: {"alarm":"hummingbird_transaction_alarm","filename":"hummingbird_transaction--localhost.yaml"}`

If you are looking for something else, let me know.  I validated the check by running our gate check script on a cloud server, validated that the check was enabled and passing in maas.  I then stopped the hummingbird services and saw the check go to red, and return to green when I started them back up.

**Comment by npawelek** [comment]
@creiht there would be a `Confd` line similar to this that says `overall success`.

**Comment by creiht** [comment]
@npawelek I see: `Wed Jul 18 08:05:12 PM 2018 INF: Confd -> config_file post overall success`

**Comment by npawelek** [comment]
@creiht Perfect, thanks! Once we get the notification plan updated in the alarm this LGTM.

File: playbooks/maas-hummingbird.yml [hunk 3]
```diff
@@ -298,6 +299,17 @@
...
+    - name: Install hummingbird transaction checks
+      template:
+        src: "templates/rax-maas/hummingbird_transaction.yaml.j2"
+        dest: "/etc/rackspace-monitoring-agent.conf.d/hummingbird_transaction--{{ inventory_hostname }}.yaml"
+        owner: "root"
+        group: "root"
+        mode: "0644"
+      delegate_to: "{{ physical_host | default(ansible_host) }}"
+      when:
+        - maas_hummingbird_transactioncheck_enabled | bool
```
Comment (npawelek) [inline playbooks/maas-hummingbird.yml]: This is false by default. Is there a reason this isn't deployed by default when hummingbird is running?

Comment (corystone) [reply playbooks/maas-hummingbird.yml]: Do you want me to set this to true?

**Comment by tonytan4ever** [comment]
LGTM, merging.

**Pull request merged**